src/decirest.erl

```diff
@@ -157,37 +108,38 @@ pretty_path(Path) when is_binary(Path) ->
 pretty_path(Path) when is_list(Path) ->
   pretty_path(iolist_to_binary(Path)).
 
--spec call_mro(_,_,#{'module':=_, 'mro':=maybe_improper_list(), _=>_}) -> {map(),_,#{'module':=_, 'mro_call':='false', _=>_}}.
-call_mro(Callback, Req, State) ->
-  call_mro(Callback, Req, State, undefined, fun(_) -> true end).
-
 -spec call_mro(_,_,#{'module':=_, 'mro':=maybe_improper_list(), _=>_},_) -> {map(),_,#{'module':=_, 'mro_call':='false', _=>_}}.
 call_mro(Callback, Req, State, Default) ->
-  call_mro(Callback, Req, State, Default, fun(_) -> true end).
+  call_mro(Callback, Req, State, Default, default).
 
 -spec call_mro(_,_,#{'module':=_, 'mro':=maybe_improper_list(), _=>_},_,_) -> {map(),_,#{'module':=_, 'mro_call':='false', _=>_}}.
-call_mro(Callback, Req, State = #{mro := MRO, module := Module}, Default, Continue) ->
-  {Res, NewReq, NewState} = call_mro(MRO, Callback, Req, State#{mro_call => true}, Default, Continue, #{}),
+call_mro(Callback, Req, State = #{mro := MRO, module := Module}, Default, ContinueType) ->
+  {Res, NewReq, NewState} = call_mro(MRO, Callback, Req, State#{mro_call => true}, Default, ContinueType, #{}),
   {Res, NewReq, NewState#{mro_call => false, module => Module}}.
 
 -spec call_mro(maybe_improper_list(),_,_,_,_,_,map()) -> {map(),_,_}.
-call_mro([{Handler, Mod} | MRO], Callback, Req0, State0, Default, Continue, Res0) ->
+call_mro([{Handler, Mod} | MRO], Callback, Req0, State0, Default, ContinueType, Res0) ->
   {ModRes, Req, State} = CallbackRes = do_callback(Handler, Callback, Req0, State0#{module => Mod}, Default),
   Res = Res0#{Mod => ModRes},
-  case Continue(CallbackRes) of
+  case continue(ContinueType, CallbackRes) of
     true ->
-      call_mro(MRO, Callback, Req, State, Default, Continue, Res);
+      call_mro(MRO, Callback, Req, State, Default, ContinueType, Res);
     false ->
       {Res, Req, State}
   end;
-call_mro([], _Callback, Req, State, _Default, _Continue, Res) ->
+call_mro([], _Callback, Req, State, _Default, _ContinueType, Res) ->
   {Res, Req, State}.
 
-continue_mro() ->
-  continue_mro(true).
-
-continue_mro(Match) ->
-  fun({Match, _, _}) -> true;(_) -> false end.
+continue(always_true, _) ->
```

Comment:
might be that we only use theese today, but by using a function that checks on the result we can do much more advanced selections and evaluations. 

It might be something we don't want to do here, but if so where would we do it instead.

Is there a performance reason for removing the function, otherwise we could do this as a pattern earlier where we create the function that we want based on an atom setting (inverted, default) or similar.

Reply:
Performance and easier api. We could add fun back as a case in continue when needed.

Maybe we could even pair {Default, ContinueType} as they are relayed.